src/include/api.h

```diff
@@ -63,10 +93,26 @@
     __oldname = (s)->name;                                                                   \
     ++(s)->api_call_counter;                                                                 \
     (s)->dhandle = (dh);                                                                     \
-    (s)->name = (s)->lastop = #struct_name "." #func_name
-#define API_SESSION_POP(s)  \
-    (s)->dhandle = __olddh; \
-    (s)->name = __oldname;  \
+    (s)->name = (s)->lastop = #struct_name "." #func_name;                                   \
+    if ((s)->id != 0 && (s)->api_call_counter == 1) {                                        \
```

Comment:
`!WT_SESSION_IS_DEFAULT(s)`.
Maybe swap the order so the check for `(s)->api_call_counter == 1` is first.  (It's more likely to be false in cases we care about in performance, when we're not using the default session).